src/main.ts

```diff
@@ -7,18 +7,25 @@ import 'bootstrap/dist/css/bootstrap.css'
 import 'bootstrap-vue/dist/bootstrap-vue.css'
 import Resource from 'vue-resource'
 import TreeView from 'vue-json-tree-view'
-
 import './registerServiceWorker'
+import 'authenticationPlugin/firebaseInit'
+import * as myAuthenticationPlugin from 'authenticationPlugin/App'
 require('./assets/css/piedpiper.css')
 Vue.config.productionTip = false
 
 Vue.use(BootstrapVue)
 Vue.use(Resource)
 Vue.use(TreeView)
-new Vue({
-  router,
-  store,
-  components: { App },
-  template: '<App/>'
-  // render: h => h(App)
-}).$mount('#app')
+
+let app: import('vue/types/vue').CombinedVueInstance<Vue, object, object, object, Record<never, any>> | undefined
+myAuthenticationPlugin.onAuthStateChanged(() => {
+  if (!app) {
+    app=new Vue({
```

Comment:
this is nor formatted. please run formatted before u do a pull request. Its best practice to run formatter after every save as a habit